assets/app/scripts/controllers/overview.js

```diff
@@ -150,9 +160,15 @@ angular.module('openshiftConsole')
             routeMap[serviceName] = routeMap[serviceName] || {};
             routeMap[serviceName][routeName] = route;
 
-            // Find the best route to display for a service. Prefer the first custom host we find.
+            $scope.routeWarningsByService[serviceName] = $scope.routeWarningsByService[serviceName] || {};
+            $scope.routeWarningsByService[serviceName][routeName] = RoutesService.getRouteWarnings(route, $scope.services[serviceName]);
+
+            // Find the best route to display for a service.
+            // Preference order: admitted custom host with TLS -> admitted custom host -> custom host -> any route 
             if (!displayRouteMap[serviceName] ||
-                (!isGeneratedHost(route) && isGeneratedHost(displayRouteMap[serviceName]))) {
+                (!isGeneratedHost(route) && isGeneratedHost(displayRouteMap[serviceName])) ||
+                ($filter("routeStatus")(route) === "Admitted" && $filter("routeStatus")(displayRouteMap[serviceName]) !== "Admitted") ||
+                (route.spec.tls && !displayRouteMap[serviceName].spec.tls)) {
```

Comment:
Bah yes just realizing the flaw in my logic

On Wed, Feb 17, 2016 at 2:24 PM, Sam Padgett <notifications@github.com>
wrote:

> In assets/app/scripts/controllers/overview.js
> <https://github.com/openshift/origin/pull/7325#discussion_r53216619>:
>
> >              if (!displayRouteMap[serviceName] ||
> > -                (!isGeneratedHost(route) && isGeneratedHost(displayRouteMap[serviceName]))) {
> > +                (!isGeneratedHost(route) && isGeneratedHost(displayRouteMap[serviceName])) ||
> > +                ($filter("routeStatus")(route) === "Admitted" && $filter("routeStatus")(displayRouteMap[serviceName]) !== "Admitted") ||
> > +                (route.spec.tls && !displayRouteMap[serviceName].spec.tls)) {
>
> Looks like it will take the last route that matches any of these
> conditions?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7325/files#r53216619>.
>